server/neptune/workflows/internal/deploy/revision/validator.go

```diff
@@ -0,0 +1,63 @@
+package revision
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/activities"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/config/logger"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deploy/terraform"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/github"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/root"
+	"go.temporal.io/sdk/workflow"
+)
+
+type githubActivities interface {
+	CompareCommit(ctx context.Context, request activities.CompareCommitRequest) (activities.CompareCommitResponse, error)
+}
+
+type Validator struct {
```

Comment:
I don't think you would even need a separate struct if your activity just returned the status as I mentioned.

Reply:
From my understanding, the worker only needs to know if the revision is valid and should be deployed right? So, I figured we could encapsulate validating the revision, and updating the check runs into `revisionValidator`  .It also makes testing easier since we don't have to worry about any of the queuing logic the worker has.